File: mouve/NodeScene.cpp
#include "NodeScene.h"
#include "NodeConnectorView.h"
#include "NodeStyle.h"

#include <QGraphicsSceneMouseEvent>
#include <QGraphicsItem>

#include <QDebug>

NodeScene::NodeScene(QObject* parent)
	: QGraphicsScene(parent)
	, mHovered(nullptr)
	, mDragging(false)
{
	setBackgroundBrush(NodeStyle::SceneBackground);
}

void NodeScene::mouseMoveEvent(QGraphicsSceneMouseEvent* event)
{
	if(mDragging)
	{
		QGraphicsItem* item = itemAt(event->scenePos());
		if(item && item->type() == NodeConnectorView::Type)
		{
			mHovered = static_cast<NodeConnectorView*>(item);
			mHovered->setHighlight(true);
		}
		else if(mHovered)
		{
			mHovered->setHighlight(false);
			mHovered = nullptr;
		}
	}
	QGraphicsScene::mouseMoveEvent(event);
}

File: mouve/ViewPrototype/NodeScene.py
from PySide.QtCore import *
from PySide.QtGui import *
from NodeView import *
from NodeLinkView import *
from NodeDataIndex import *
from NodeConnectorView import *

class NodeScene(QGraphicsScene):

	dragging = False
	hovered = None

	def __init__(self, parent=None):
		QGraphicsScene.__init__(self, parent)
		self.setBackgroundBrush(NodeStyle.SceneBackground)

	def mouseMoveEvent(self, event):
		if self.dragging:
			item = self.itemAt(event.scenePos())
			if item and item.type() == NodeConnectorView.Type:
				self.hovered = item
				self.hovered.setHighlight(True)				
			elif self.hovered:
				self.hovered.setHighlight(False)
				self.hovered = None
		QGraphicsScene.mouseMoveEvent(self, event)